src/java.desktop/share/classes/javax/swing/JPasswordField.java

```diff
@@ -544,8 +543,7 @@ public String getAtIndex(int part, int index) {
          */
         public String getAfterIndex(int part, int index) {
             if (part == AccessibleText.CHARACTER) {
-                String str = super.getAfterIndex(part, index);
-                return getEchoString(str);
+                return getEchoString(super.getAfterIndex(part, index));
```

Comment:
I don't see how removing the local variable changes anything. Explanation ??

Reply:
Here it is just a slight code cleanup. We do not need additional variable for passing value from one method to another. It serves no other purpose at all.

Reply:
Let me ask it this way.
Does super.getAfterIndex(part, index) return a String with any of the password in clear text ? 
It seems to me like it might.

Reply:
For CHARACTER it will return String with a single character in the corresponding position. There is a possibility that someone will iterate the entirety of the password text and get all the characters in the password as a separate strings but digging it from the memory dump is much more difficult than the singular string with the whole password.
For anything but character we do not use this method - we get password as an array of characters and - after the fix - immediately overriding them with the same number of echo characters.